drivers/video/video.c

```diff
@@ -705,13 +710,26 @@ static void change_video_state(FAR video_mng_t    *vmng,
               video_framebuff_get_vacant_container(&vmng->video_inf.bufinf);
       if (container)
         {
-          start_capture(V4L2_BUF_TYPE_VIDEO_CAPTURE,
-                        vmng->video_inf.nr_fmt,
-                        vmng->video_inf.fmt,
-                        &vmng->video_inf.clip,
-                        &vmng->video_inf.frame_interval,
-                        container->buf.m.userptr,
-                        container->buf.length);
+          if (container->buf.memory == V4L2_MEMORY_MMAP)
```

Comment:
why do you need check V4L2_MEMORY_MMAP? userptr already point to the right place.